### ISSUE TITLE
Request Feature OC switching

### ISSUE THREAD

**Issue by Wapp92** [aaronsace/MultiPoolMiner#724]
Hi aaronsace, do you plan to add a overclock switching feature?

For me lyra2z is unstable with equihash OC, and equihash is underperfoming with Lyra2z setting.
Do you think its possible to switch between afterburner profile when a certain algo is running. Or any other way. 

Thanks

**Comment by aaronsace** [comment]
WattTool profiles is the plan. 

https://github.com/aaronsace/MultiPoolMiner/projects/1

**Comment by aaronsace** [comment]
I recommend you focus your OC to increase stability across all algorithms that you are using.

**Comment by Wapp92** [comment]
Thanks aaronsace, can wait for WattTool.
My biggest stability issue is with lyra2z i just can it to work even with no OC, i have to underclock memory so much that it really hurts other algo. So for now i just remove Lyra2z.

**Comment by fonyo** [comment]
I agree with @aaronsace. If you are on the edge with your OC of your cards, you may not notice but your cards will generate errors which means you halve your hashrate for smaller periods As your cards need to redo the work. This is not about getting incorrect shares, it happens internally. Sometimes less is more and you also increase stability and lifespan of your cards while reducing the heat output.

**Comment by Wapp92** [comment]
That s really tricky, how can you optimize oc if you don t see any error. I need time to test each memory setting for Lyra2z to find what is stable, this should get me a stable overall oc. But if i need to thing about error generate that don t show, it s a endless loop.

**Comment by aaronsace** [comment]
MultiPoolMiner doesn't support OC at the moment. I personally just drop the core voltage slightly and increase the fans.

**Comment by FabianGabor** [comment]
There's a simple solution for Nvidia users. With MSI Afterburner it is really easy to load presets with the command:

"{Path}\MSIAfterburner.exe" -Profile{1-5}

This is really cool and would be nice to have a default profile that would be used when launching MPM and for certain algos load a different one. For example Ethash is stable even at memory speeds much higher than other algos would be and has the same performance between 72-100% power limits so it makes sense to lower the power usage to reduce costs.

**Comment by mic4rd** [comment]
Quick question for help: Changing the start.bat like this:

> cd /d %~dp0
> 
> setx GPU_FORCE_64BIT_PTR 1
> setx GPU_MAX_HEAP_SIZE 100
> setx GPU_USE_SYNC_OBJECTS 1
> setx GPU_MAX_ALLOC_PERCENT 100
> setx GPU_SINGLE_ALLOC_PERCENT 100
> 
> ::Load mining profile
> "C:\Program Files (x86)\MSI Afterburner\MSIAfterburner.exe" -Profile5
> 
> set "command=& .\multipoolminer.ps1 -wallet ...
> 
> pwsh -noexit -executionpolicy bypass -windowstyle maximized -command "%command%"
> powershell -version 5.0 -noexit -executionpolicy bypass -windowstyle maximized -command "%command%"
> msiexec -i https://github.com/PowerShell/PowerShell/releases/download/v6.0.1/PowerShell-6.0.1-win-x64.msi -qb!
> pwsh -noexit -executionpolicy bypass -windowstyle maximized -command "%command%"
> 
> pause
> 
> ::Load standard/gaming profile
> "C:\Program Files (x86)\MSI Afterburner\MSIAfterburner.exe" -Profile1`
> 

loads the Underclocked profile jist fine before starting MPM, but it doesn't switch back after closing MPM. Anyone knows what has to be changed?
